test/integration/ClientNotificationMongoRepoSpec.scala

```diff
@@ -0,0 +1,218 @@
+  private val validClientSubscriptionId2 = ClientSubscriptionId(validClientSubscriptionId2UUID)
+
+  private val validConversationIdString: String = "638b405b-9f04-418a-b648-ce565b111b7b"
+  private val validConversationIdStringUUID = UUID.fromString(validConversationIdString)
+  private val validConversationId = ConversationId(validConversationIdStringUUID)
+
+  private val payload1 = "<foo1></foo1>"
+  private val payload2 = "<foo2></foo2>"
+  private val payload3 = "<foo3></foo3>"
+
+  private val headers = Seq(Header("h1","v1"), Header("h2", "v2"))
+  private val notification1 = Notification(validConversationId, headers, payload1, CustomMimeType.XmlCharsetUtf8)
+  private val notification2 = Notification(validConversationId, headers, payload2, CustomMimeType.XmlCharsetUtf8)
+  private val notification3 = Notification(validConversationId, headers, payload3, CustomMimeType.XmlCharsetUtf8)
+
+  private val client1Notification1 = ClientNotification(validClientSubscriptionId1, notification1)
+  private val client1Notification2 = ClientNotification(validClientSubscriptionId1, notification2)
+  private val client1Notification3 = ClientNotification(validClientSubscriptionId1, notification3)
+  private val client2Notification1 = ClientNotification(validClientSubscriptionId2, notification1)
+
+  private val mockNotificationLogger = mock[NotificationLogger]
+  private val mockErrorHandler = mock[ClientNotificationRepositoryErrorHandler]
+
+  private lazy implicit val emptyHC: HeaderCarrier = HeaderCarrier()
+  private val timeoutInSeconds = 2
+  private val duration = org.joda.time.Duration.standardSeconds(timeoutInSeconds)
+
+  private val mongoDbProvider = new MongoDbProvider {
+    override val mongo: () => DB = self.mongo
+  }
+
+  val lockRepository = new LockRepository
+  val lockRepo: LockRepo = new LockRepo() {
+    val db: () => DB = () => mock[DB]
+    override val repo: LockRepository = lockRepository
+  }
+
+  private val repository = new ClientNotificationMongoRepo(mongoDbProvider, lockRepo, mockErrorHandler, mockNotificationLogger)
+
+  override def beforeEach() {
+    await(repository.drop)
+    await(lockRepository.drop)
+  }
+
+  override def afterAll() {
+    await(repository.drop)
+    await(lockRepository.drop)
+  }
+
+  private def collectionSize: Int = {
+    await(repository.collection.count())
+  }
+
+  private def selector(clientSubscriptionId: ClientSubscriptionId) = {
+    Json.obj("csid" -> clientSubscriptionId.id)
+  }
+
+  "repository" should {
+    "successfully save a single notification" in {
+      when(mockErrorHandler.handleSaveError(any(), any(), any())).thenReturn(true)
+      val saveResult = await(repository.save(client1Notification1))
+      saveResult shouldBe true
+      collectionSize shouldBe 1
+
+      val findResult = await(repository.collection.find(selector(validClientSubscriptionId1)).one[ClientNotification]).get
+      findResult._id should not be None
+      findResult.timeReceived should not be None
```

Comment:
would be cleaner to extract `PassByNameVerifier(...` code to a helper function that is reused

Reply:
Done.